### PR TITLE
feat(provider/kubernetes): Properly version all versioned resources

### PR DESCRIPTION
**Pull request by lwander**
Now any resource that's flagged as versioned in k8s will be given a `-vN` suffix

### CODE REVIEW

**Comment by lwander** [comment]
spinnaker/spinnaker#1902

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java [hunk 1]
```diff
@@ -51,4 +61,38 @@ public KubernetesCoordinates toCoordinates(Artifact artifact) {
...
+          }
+        } )
+        .filter(Objects::nonNull)
+        .filter(i -> i >= 0)
+        .collect(Collectors.toList());
+
+    taken.sort(Integer::compareTo);
+
+    int attempt = 0;
+    for (Integer exists : taken) {
```
Comment (jtk54) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java]: Could be a `forEach`.

**Comment by jtk54** [comment]
Needs a little re-base.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java (hunk 1 above)
Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java]: I don't really see the advantage, unless we could just plug a little stateless lambda in.

Comment (jtk54) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java]: Just Java 8 style is all. Fine as it is.

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java]: https://stackoverflow.com/a/20177092/1435439

Comment (jtk54) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java]: Rescinded.

**Pull request merged**